docs/extensions/controlplane-webhooks.md

```diff
@@ -28,6 +28,8 @@ You can install 3 different kinds of controlplane webhooks:
 
 The labels `shoot.gardener.cloud/provider` and `seed.gardener.cloud/provider` are added by Gardener when it creates the Shoot namespace.
 
+To reduce controlplane webhook calls for unrelated changes, the label `provider.extensions.gardener.cloud/mutated-by-controlplane-webhook: true` can be added to targeted resources. The controlplane webhooks can use this label to explicitly specify resources to target using an `ObjectSelector` in the constructor. 
```

Comment:
The generic controlplane mutator mutates also ETCD, used by the controlplaneexposure webhooks: https://github.com/gardener/gardener/blob/dcf0191b4f2a883f9115fd8316bc6884316abdf6/extensions/pkg/webhook/controlplane/genericmutator/mutator.go#L170-L184 and https://github.com/gardener/gardener-extension-provider-aws/blob/v1.55.1/pkg/webhook/controlplaneexposure/ensurer.go#L34-L67.

Will it be tackled in this PR or in another PR?

Reply:
As the configuration for using the selector is done when adding the webhook to the manager and the configuration is webhook-specific, I would suggest to handle this case in the according extension(s): https://github.com/gardener/gardener-extension-provider-aws/blob/619625f5078f036454b58b7cc157626c54191efe/pkg/webhook/controlplaneexposure/add.go#L41

There are no modifications required in the generic mutator for this to work anyway.

Reply:
After offline discussion with @LucaBernstein : there is no need of object selector in the controlplane exposure webhook. There are only two Etcd resources in the Shoot control plane namespace in the Seed and the exposure webhook has to mutate both of them.